### ISSUE TITLE
venv-enabled-x86_64.txt unable to download 

### ISSUE THREAD

**Issue by Luiginoo007** [uyuni-project/uyuni#5739]
# Problem description

Uyuni setup complete, bootstrap scripts generated.
When I am trying to bootstrap first client I am all the time getting this error :
Error: Unable to download https://myuyuniserverurl:443/pub/repositories/sle/15/3/bootstrap/venv-enabled-x86_64.txt file!

## Version of Uyuni Server and Proxy (if used)

Information for package Uyuni-Server-release:
---------------------------------------------
Repository     : UYUNI repository
Name           : Uyuni-Server-release
Version        : 2022.06-183.9.uyuni2
Arch           : x86_64
Vendor         : obs://build.opensuse.org/systemsmanagement:Uyuni
Support Level  : Level 3
Installed Size : 1.4 KiB
Installed      : Yes (automatically)
Status         : up-to-date
Source package : Uyuni-Server-release-2022.06-183.9.uyuni2.src
Summary        : Uyuni Server

Proxy not in use

Bootstraping new installation of SLES15sp3

## Details about the issue

rhn_web_ui.log :
Error: Unable to download https://myuyuniserverurl:443/pub/repositories/sle/15/3/bootstrap/venv-enabled-x86_64.txt file!

api log from salt 
2022-08-04 01:22:16,785 [salt.client.ssh  :1207][ERROR   ][7707] Error running ssh_pre_flight script preflight.sh for host '--clientIP--'

Fix from here https://github.com/uyuni-project/uyuni/issues/5372 is already included in my Uyuni version 

when I run mgr-create-bootstrap-repo
I am getting 
ERROR: package 'python3-hwdata' not found
ERROR: package 'python3-rhnlib' not found
ERROR: package 'spacewalk-client-tools' not found
ERROR: package 'python3-spacewalk-client-tools' not found
ERROR: package 'suseRegisterInfo' not found
ERROR: package 'python3-suseRegisterInfo' not found
ERROR: package 'venv-salt-minion' not found

but all of these packages are present on my Uyuni server

**Comment by Luiginoo007** [comment]
After long searching I found that I need uyuni client channel to sync. 
In my case of SLE15sp3 : spacewalk-repo-sync --channel sles15-sp3-uyuni-client-x86_64
then I need to bootstrap it and uyuni client has to be part of it 
mgr-create-bootstrap-repo -c SLE-15-SP3-x86_64 -d --with-custom-channels --with-parent-channel=sles15-sp3-uyuni-client-x86_64 

I hope that this will help someone. I will see how it will be in GUI, now I cannot use it to sync because GUI is ignoring proxy setting to sync repo

**Comment by Luiginoo007** [comment]
I have still issue with SLES12SP5 

spacewalk-repo-sync --channel sles12-sp5-uyuni-client-x86_64
is not working, I am getting Channel sles12-sp5-uyuni-client-x86_64 is not custom or does not exist.

then I cannot use it in 
mgr-create-bootstrap-repo -c SLE-12-SP5-x86_64 -d --with-custom-channels --with-parent-channel=sles12-sp5-uyuni-client-x86_64

**Comment by mcalmer** [comment]
`--with-parent-channel=sles12-sp5-uyuni-client-x86_64`
I think sles12-sp5-uyuni-client-x86_64 is not the parent channel. If you have mirrored SLES12 SP5 from SCC, it should be something like  `sles12-sp5-pool-x86_64` . But maybe you do not need this option at all. AFAIK it is only needed when multiple are available which may not be the case. Depends on what you mirrored. AFAIK there is no SLED (Desktop) for SLE12, so that option is already gone. SLE15 has more cases where this is needed.

But anyway. What are the logs saying? Please run it in debug mode (`--debug`) and provide /var/log/rhn/mgr-create-bootstrap-repo/mgr-create-bootstrap-repo.log

**Comment by Luiginoo007** [comment]
Thanks for reply

First of all when I tried to create boostrap repo mgr-create-bootstrap-repo --channel SLE-12-SP5-x86_64 I was getting similar errors as I described  in first article : ERROR: package 'venv-salt-minion' not found ... and more 

yes you are right that it is not parental channel, but is has to be part of channel SLES12-SP5-Pool for x86_64. If not I get that ERRORs 

In the mean time I was able to create bootstraprepo. I am still beginner with uyuni so I an not sure if this is correct way how to do it.
1. in GUI - Software -Manage- Repositories I created new repository :External - Uyuni Client Tools for SLES12 SP5 x86_64
 
![image](https://user-images.githubusercontent.com/56265902/183971146-7de9ebd6-0a57-4b41-9625-ee3f96c19a0c.png)

as a samle I used already created SLES15 SP5
repository URL is https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Stable:/SLE12-Uyuni-Client-Tools/SLE_12/ 

2. I created in GUI - Software -Manage- Channel new with name SLES12-SP5-Pool for x86_64 

![image](https://user-images.githubusercontent.com/56265902/183972067-7feba511-74c3-41a5-af48-9b721e011d0f.png)

I fill it similar as I have for SLES15 ...
In tab repositories I marked previously created repository : External - Uyuni Client Tools for SLES12 SP5 x86_64
Here, one detail is missing in SLES12 channel and that is GPG key ID and key fingerprint, probably because I created it manually. in SLES15 channel it is not missing because I was able to sync it via cli.

after that I was able to do spacewalk-repo-sync --channel sles12-sp5-uyuni-client-x86_64 
and 
mgr-create-bootstrap-repo -c SLE-12-SP5-x86_64 -d --with-custom-channels --with-parent-channel=sles12-sp5-uyuni-client-x86_64

without errors 

now Uyuni Client Tool is part of my SLES12-SP5-Pool channel : 
![image](https://user-images.githubusercontent.com/56265902/183973904-8ef4e91d-5bab-466b-9b98-5a68c67caee1.png)
